components/template/template-application-rest/src/main/resources/META-INF/dirigible/template-application-rest/api/entity.ts.template

```diff
@@ -210,17 +210,32 @@ class ${name}Service {
     private checkPermissions(operationType: string) {
 #if($perspectiveRole)
         if (!user.isInRole("${perspectiveRole}")) {
-            throw new ForbiddenError();
+            const perspectiveRoles = [
+                "${perspectiveRole}",
+            ];
+
+            throw new ForbiddenError(perspectiveRoles);
         }
 #end
 #if($roleRead)
-        if (operationType === "read" && !user.isInRole("${roleRead}")) {
-            throw new ForbiddenError();
+        if (operationType === "read" && !(user.isInRole("${roleRead}")#if($roleWrite) || user.isInRole("${roleWrite}")#end)) {
+            const readRoles = [
+                "${roleRead}",
+                #if($roleWrite)
+                "${roleWrite}",
+                #end
+            ];
+            
+            throw new ForbiddenError(readRoles, "read");
         }
 #end
 #if($roleWrite)
         if (operationType === "write" && !user.isInRole("${roleWrite}")) {
-            throw new ForbiddenError();
+            const writeRoles = [
+                "${roleWrite}",
+            ];
+            
+            throw new ForbiddenError(writeRoles, "write");
```

Comment:
From Security PoV I'm not sure that it's a good idea to tell the user, which role is missing. Let's leave it as it was.